cpu/cc2538/dev/cc2538-rf.c

```diff
@@ -995,10 +1070,20 @@ void
 cc2538_rf_rx_tx_isr(void)
 {
   ENERGEST_ON(ENERGEST_TYPE_IRQ);
+  
+#if CC2538_RF_CONF_SFD_TIMESTAMPS
+  if(poll_mode) {
+    if(REG(RFCORE_XREG_FSMSTAT1) & RFCORE_XREG_FSMSTAT1_RX_ACTIVE) {
+      cc2538_sfd_rtime = RTIMER_NOW();
+    }
+  }
+#endif
```

Comment:
Have you looked at the MAC timer (section 12) of the UG? It can be used to automatically capture SFD, so you don't have to enable the SFD interrupt. You would probably need to convert this capture value to an rtimer representation.

The SFD interrupt will also get raised after TX of an SFD. I cannot see any attempt at determining whether it was an RX or a TX that generated the SFD interrupt. Are you handling this anywhere that I cannot see?